src/huggingface_hub/hf_api.py

```diff
@@ -1579,16 +1579,19 @@ def save_token(cls, token):
             f.write(token)
 
     @classmethod
-    def get_token(cls):
+    def get_token(cls) -> Optional[str]:
         """
-        Get token or None if not existent.
+        Get token or None if not existent. A token can be also provided using env variables: `export HUGGING_FACE_HUB_TOKEN=<YOUR_TOKEN>`.
         """
+        token: Optional[str] = os.environ.get("HUGGING_FACE_HUB_TOKEN")
```

Comment:
Don't we want to use the token if provided?

```suggestion
        token: Optional[str] = os.environ.get("HUGGING_FACE_HUB_TOKEN")
        if token:
            return token
```

Reply:
changed with a check `if token is None` to avoid early returns